cmd/packaging/packaging.go

```diff
@@ -162,15 +157,166 @@ func runDockerPackaging(path string, packagingFormat string, command []string) {
 	}
 }
 
+var templateData map[string]string
+var once sync.Once
+
 func getTemplateData(projectName, buildVersion string) map[string]string {
-	return map[string]string{
-		"projectName":         projectName,
-		"strippedProjectName": removeDashesAndUnderscores(projectName),
-		"author":              getAuthor(),
-		"version":             buildVersion,
-		"description":         pubspec.GetPubSpec().Description,
-		"debDependencies":     strings.Join(linuxPackagingDependencies, ","),
-		"snapDependencies":    strings.Join(linuxPackagingDependencies, "\n      - "),
-		"organizationName":    androidmanifest.AndroidOrganizationName(),
+	once.Do(func() {
+		templateData = map[string]string{
+			"projectName":         projectName,
+			"strippedProjectName": strings.ReplaceAll(projectName, "_", ""),
+			"author":              getAuthor(),
+			"version":             buildVersion,
+			"description":         pubspec.GetPubSpec().Description,
+			"organizationName":    androidmanifest.AndroidOrganizationName(),
+			"arch":                runtime.GOARCH,
+		}
+	})
+	return templateData
+}
+
+type packagingTask struct {
+	packagingFormatName            string
+	dependsOn                      map[*packagingTask]string
+	templateFiles                  map[string]string
+	executableFiles                []string
+	linuxDesktopFileExecutablePath string
+	linuxDesktopFileIconPath       string
+	dockerfileContent              []string
+	generateBuildFiles             func(projectName, path string)
+	buildOutputDirectory           string
+	packagingCommand               string
+	resultFileName                 string
+	outputFileExtension            string
```

Comment:
I think it would help future developers on this code a lot to have comments on these fields explaining what they're for and how they should be used.